feat: implement user authentication system with JWT and role-based access control

baiyaoshi opened the pull request:
Implement login and registration with JWT utility enhancements

fix: #8

Hunger61 committed:
feat:Implement obtaining current user profile.

Hunger61 committed:
feat:Implement obtaining current user profile.

Hunger61 committed:
feat:Implement obtaining current user profile.

baiyaoshi committed:
feat:ImplementGettingValidateImage

Hunger61 committed:
feat:Implement Obtaining the required information form to be submitted

baiyaoshi committed:
feat:Implement getting validate code

baiyaoshi committed:
feat:Implement registration function
fix:jwtUtil

baiyaoshi committed:
feat:Implement login function

baiyaoshi committed:
feat:add logining registering and getting validate code

Hunger61 committed:
feat:Implement Obtaining the summited information form to be submitted.
refactor:Optimize dependency injection in the service layer

baiyaoshi committed:
feat: implement login and registration function

baiyaoshi committed:
chore: restrict GitHub token permissions in apifox.yml

baiyaoshi committed:
fix:resolve conflict

RooobinYe committed:
Merge branch 'dev' into feat/implement-login-function

Hunger61 committed:
refactor:Using constructor injection instead of field injection.

RooobinYe committed:
feat: enhance GitHub Actions to start Spring Boot app with database

- Add MySQL and Redis services to GitHub Actions
- Create application.yml dynamically for CI environment
- Build and start Spring Boot application before running API tests
- Add health check to ensure application is ready
- Update Node.js version and improve workflow structure

RooobinYe committed:
feat: add database schema and initial data

- Add schema.sql with user table structure
- Add data.sql with default admin user
- Support automatic database initialization

RooobinYe committed:
fix: improve GitHub Actions multi-module build process

- Fix Maven multi-module dependency resolution by building entire project first
- Separate build and start steps for better error handling
- Add proper application startup with logging redirection
- Fix SQL file formatting with proper newlines

baiyaoshi committed:
feat: initial implementation of OSS functionality, added file table SQL statement

baiyaoshi committed:
fix: resolve merge conflicts in schema.sql and data.sql

RooobinYe committed:
fix: add sql and add todo in LoginController.java

Hunger61 commented on approval-pojo/src/main/java/fun/sast/dto/RegisterUserDTO.java:
名词+动词/形容词+DTO，改一下

Bai-Yao-Shi commented on approval-pojo/src/main/java/fun/sast/entity/Token.java:
要，暂时先留着吧

Hunger61 commented on approval-pojo/src/main/java/fun/sast/dto/LoginUserDTO.java:
而且怎么没有validateCode

Bai-Yao-Shi replied:
注册接口文档没有我自己用的

RooobinYe commented on approval-pojo/src/main/java/fun/sast/dto/RegisterUserDTO.java:
controller 中的 register 接口也需要移除

Hunger61 commented:
我写了response的全局处理，只需要在方法上加@ResponseResult就可以自动包装成GlobalResponse

Hunger61 commented on approval-server/pom.xml:
同样

Hunger61 commented on approval-server/src/main/java/fun/sast/controller/publicController/FileController.java:
规范方法命名，动词+名词

Hunger61 commented:
在每个方法上的注释，最好使用标准JavaDoc格式
/**
 * 方法描述（简洁明了地说明方法的作用）
 * 
 * @param 参数名 参数描述
 * @return 返回值描述
 * @throws 异常类型 异常描述
 */

Hunger61 committed:
refactor:Standard Annotations.

Hunger61 committed:
refactor:Standardized format.

Hunger61 committed:
Merge branch 'feat/file-upload-and-submission-APIs#13' into master

baiyaoshi committed:
fix: resolve architecture issues and improve code quality

baiyaoshi committed:
update getDownloadCertificate method

baiyaoshi committed:
feat: implement competition notice retrieval

RooobinYe commented:
所有的功能都写好了吗？如果还没有的话先 draft 哈

baiyaoshi committed:
fix: modify part of logic in verification code and login process

baiyaoshi commented:
> 所有的功能都写好了吗？如果还没有的话先 draft 哈

我push了一下这里的draft会没吗🤔

baiyaoshi commented:
> 所有的功能都写好了吗？如果还没有的话先 draft 哈

才发现拦截器还没拉进来，加上这个基本上可以了大概

RooobinYe commented:
> > 所有的功能都写好了吗？如果还没有的话先 draft 哈
> 
> 我 push 了一下这里的 draft 会没吗🤔

不会

baiyaoshi committed:
chore: merge master into feat/implement-login-function

baiyaoshi committed:
fix:update schema.sql and data.sql

baiyaoshi commented:
> > > 所有的功能都写好了吗？如果还没有的话先 draft 哈
> > 
> > 
> > 我 push 了一下这里的 draft 会没吗🤔
> 
> 不会

呃这个拦截器好像还没做我不好加了，除此之外的功能好像差不多了

baiyaoshi committed:
feat: add basic user interceptor and access control

RooobinYe commented:
获取比赛列表 如果是未登录的用户 应该可以获取普通学生（role=0）的公告

RooobinYe committed:
feat: configure OSS file storage and update interceptor exclusions

- Add OSS configuration for file storage in dev environment
- Update WebConfig to exclude additional public endpoints from authentication
- Fix notice table insert with explicit ID values in data.sql
- Update GitHub workflow with OSS mock configuration

baiyaoshi committed:
fix: remove debug logs, update notice logic for unauthenticated users, and include /com/notice/list in interceptor paths

RooobinYe committed:
refactor: remove unused UserRegisterDTO and update interceptor exclusions

baiyaoshi commented:
/com/notice/list要从拦截器解析身份🤔

RooobinYe committed:
feat: enable optional authentication for /com/notice/list endpoint

- Remove /com/notice/list from interceptor exclusion paths in WebConfig
- Update UserInterceptor to support optional authentication for notice list endpoint
- Allow unauthenticated users to access notice list while still parsing identity if token is provided
- Maintain backward compatibility for other endpoints requiring authentication

RooobinYe committed:
feat: add PR celebration workflow and misc updates

- Add GitHub Actions workflow for PR merge celebrations via Feishu bot
- Fix comment formatting in RedisUtil
- Update notice test data with published status
- Fix mybatis-plus version reference in pom.xml

RooobinYe commented on approval-server/src/main/java/fun/sast/controller/publicController/CommonController.java:
```diff
@@ -1,36 +1,32 @@
-    public Map<String, String> downloadCertificate(@RequestParam String url) {
-        return new HashMap<>();
+    public GlobalResponse noticeList(@RequestParam String id) {
+        List<Notice> noticeList = noticeService.getNotices(id);
```
这里使用 `List<NoticeVO>` 是不是更好一点？ @Hunger61 你觉得呢？

包括其他地方的 VO 和 DTO 用的不是很规范

Hunger61 replied:
是的

RooobinYe reviewed (approved):
LGTM 👏

其他的一些地方有点小问题 我觉得问题不是很大

我想要先这部分合进去 这样其他人也可以开始写了  这部分有问题可以再提 PR fix 就行

baiyaoshi commented:
我点合并就行了吗

RooobinYe commented:
> 我点合并就行了吗
> 
> 

你可以看到 merge 吗？很奇怪啊 你应该没有权限的🤔

你不需要操作任何内容 接下来是等 @Hunger61 code review 之后由我和他中的一位进行合并

为了保证线性的历史记录 我的建议是使用 `Squash Merge`

baiyaoshi commented:
好，我这里是commit merge

RooobinYe commented:
> 好，我这里是commit merge

好的 这应该是我权限设置的问题 明天我修一下

RooobinYe commented:
@Hunger61 Please review.

RooobinYe committed:
docs: improve JavaDoc comments for UserService login method

Hunger61 commented on approval-common/src/main/java/fun/sast/utils/FileUtil.java:
是否考虑传入的路径为空时导致崩溃。

Hunger61 commented on approval-common/src/main/java/fun/sast/utils/OSSUtil.java:
建议使用log，别用System.out
